ui/views.py

```diff
@@ -56,6 +58,26 @@ def default_js_settings(request):
     }
 
 
+def conditional_response(view, *args, **kwargs):
+    """
+    Redirect to login page if user is anonymous and video is private.
+    Raise a permission denied error if user is logged in but doesn't have permission.
+    Otherwise, return standard template response.
+
+    Args:
+        view(TemplateView): a video-specific View object (ViewDetail, ViewEmbed, etc).
+
+    Returns:
+        TemplateResponse: the template response to render.
+    """
+    if not ui_permissions.has_video_view_permission(view.video, view.request):
+        if view.request.user.is_authenticated():
+            raise PermissionDenied
+        else:
+            return redirect('{}?next={}'.format(settings.LOGIN_URL, view.request.path))
```

Comment:
Can you use `django.contrib.auth.views.redirect_to_login` instead here?

Reply:
✅